### PR TITLE
Add single entrypoint proxy interface documentation

### PR DESCRIPTION
**Pull request by pwplusnick**
## Description
Several implementations of this proxy contract exist for multiple
different runtimes. In effort to document this existing code, these
changes have been put down as part of the documentation for OpenWhisk
since they do not properly fit under the definition of any one runtime.
In addition, a small footer has been added to the docs/action-new.md to
make note of and link to the docs/single_entrypoint_proxy_contract.md describing
and giving examples of this contract.

## Related issue and scope

n/a
## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [x] Documentation

## Types of changes

- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [x] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [x] I updated the documentation where necessary.

### CODE REVIEW

File: docs/single_entrypoint_proxy_contract.md [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+{
+  "payload": "Hello Alan Turing from England!"
+}
+```
+
+## Example Cases
+Below is a table outlining the standardized behaviors that any action proxy implementation needs to fulfill. NodeJS was the sample language used, but corresponding example cases could be written in the language of the corresponding runtime it is showcasing.
+
+
+<table border="2" cellspacing="0" cellpadding="6" rules="groups" frame="hsides">
```
Comment (rabbah) [inline docs/single_entrypoint_proxy_contract.md]: can you use a markdown table instead?

**Comment by pwplusnick** [comment]
https://github.com/apache/openwhisk/pull/4850/commits/31df9a6e13a6fbbf0530e944b804d8abc8a3bb97
Note this is the version you should look at for file changes. The last few commits were just me trying to get a markdown table working, but it doesn't look like it will pan out.

**Comment by rabbah** [comment]
Thanks for considering it and trying.

File: docs/single_entrypoint_proxy_contract.md [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+
+<col/>
+
+<col/>
+</colgroup>
+<thead>
+<tr>
+<th scope="col" >Test Name</th>
+<th scope="col" >Action Code (In NodeJS)</th>
+<th scope="col" >Input</th>
```
Comment (mrutkows) [inline docs/single_entrypoint_proxy_contract.md]: Can we please link to the JSON files since they are now merged into openwhisk-test?

Comment (pwplusnick) [reply docs/single_entrypoint_proxy_contract.md]: They are linked further down in the Nodejs table

**Comment by mrutkows** [comment]
@pwplusnick Will, you may want to circle back with Michele and team to invite them to make follow-on changes as it seems they may have been too busy in the last week to review this specific docs. PR.

**Pull request merged**